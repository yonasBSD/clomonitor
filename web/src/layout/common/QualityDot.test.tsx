import { render, screen } from '@testing-library/react';
import { vi } from 'vitest';

import QualityDot from './QualityDot';
import styles from './QualityDot.module.css';

describe('QualityDot', () => {
  afterEach(() => {
    vi.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<QualityDot level={1} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<QualityDot level={1} />);

    expect(screen.getByTestId('quality-dot')).toBeInTheDocument();
    expect(screen.getByTestId('quality-dot')).toHaveClass(styles.level1);
  });

  it('renders level 3 dot', () => {
    render(<QualityDot level={3} />);

    expect(screen.getByTestId('quality-dot')).toHaveClass(styles.level3);
  });
});
