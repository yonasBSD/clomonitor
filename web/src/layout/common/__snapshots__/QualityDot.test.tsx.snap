// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QualityDot > creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="me-2 border border-1 rounded-pill position-relative _quality_d52a11 _level1_d52a11 undefined"
    data-testid="quality-dot"
  />
</DocumentFragment>
`;
